### ISSUE TITLE
Thread panic upon service of files

### ISSUE THREAD

**Issue by tcullum-gpsw** [thecoshman/http#76]
Every time this crate serves a file, I get this:
```
[2018-12-23 01:16:27] ip was served file /home/moveax41h/static_server/index.html as text/html
[2018-12-23 01:16:27]                     found index file for directory /home/moveax41h/static_server
thread '<unnamed>' panicked at 'Failed to get file metadata: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
[2018-12-23 01:16:45] ip was served file /home/moveax41h/static_server/index.html as text/html
[2018-12-23 01:16:45]                     found index file for directory /home/moveax41h/static_server
thread '<unnamed>' panicked at 'Failed to get file metadata: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:1009:5
```

Using nightly rust rustc 1.33.0-nightly (2d3e909e4 2018-12-22)

Server still works, but just a weird occurrence.

**Comment by nabijaczleweli** [comment]
Yeah, I've been eyeing the `panic!()`king issue for a long time but I've resented going back into the codebase because it'd entail rewriting a bunch of error handling which I'm not looking forward to :v

That particular one looks like it's coming across an inaccessible file during directory enumeration.

I'll try to take a look at any cases of this particular error in the next couple of days, though.

**Comment by nabijaczleweli** [comment]
@tcullum-gpsw Do please check out the `develop` branch to see if the issue has been fixed and, if so, we'll push out a release.

**Comment by tcullum-gpsw** [comment]
@nabijaczleweli Hey sorry for the delay, been busy around here.

Problem solved... Just tried building/running the develop branch and no more thread panics!

```
[2018-12-31 15:04:41] 127.0.0.1:64128 was served file /Users/tcullum/rust/http/LICENSE as text/plain
[2018-12-31 15:04:41] Created temp dir $TEMP/http-Users-tcullum-rust-http/encoded
[2018-12-31 15:04:41]                 encoded as gzip for 164.9% ratio
[2018-12-31 15:04:41] 127.0.0.1:64128 requested to GET nonexistant entity /Users/tcullum/rust/http/favicon.ico
[2018-12-31 15:04:41]                 encoded as gzip for 519.4% ratio
[2018-12-31 15:04:55] 127.0.0.1:64130 was served file /Users/tcullum/rust/http/build.rs as text/x-rust
[2018-12-31 15:04:55]                 encoded as gzip for 270.1% ratio
```

**Comment by nabijaczleweli** [comment]
That's good to hear! Unfortunately, though, Pirate is away, and so we cannot push out a crates.io release 'til he comes back in two weeks' time :v

**Comment by nabijaczleweli** [comment]
Released in [`v1.4.1`](https://github.com/thecoshman/http/releases/tag/v1.4.1).
